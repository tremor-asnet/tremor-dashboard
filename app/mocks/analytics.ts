export const SALES_BY_COUNTRY = [
  {
    id: "us",
    flag: "/assets/images/flags/US.webp",
    country: "United State",
    sales: "2500",
    value: "$230,900",
    bounce: "29.9%",
  },
  {
    id: "ge",
    flag: "/assets/images/flags/DE.webp",
    country: "Germany",
    sales: "3.900",
    value: "$440,000",
    bounce: "40.22%",
  },
  {
    id: "gb",
    flag: "/assets/images/flags/GB.webp",
    country: "Great Britain",
    sales: "1.400",
    value: "$190,700",
    bounce: "23.44%",
  },
  {
    id: "br",
    flag: "/assets/images/flags/BR.webp",
    country: "Brasil",
    sales: "562",
    value: "$143,960",
    bounce: "32.14%",
  },
  {
    id: "au",
    flag: "/assets/images/flags/AU.webp",
    country: "Australia",
    sales: "400",
    value: "$143,960",
    bounce: "56.83%",
  },
];

export const STATISTICAL_DATA = [
  {
    id: "1",
    type: "Bookings",
    amount: 281,
    totalAmount: 55,
    duration: "than last week",
  },
  {
    id: "2",
    type: "Today's Users",
    amount: 2300,
    totalAmount: 3,
    duration: "than last month",
  },
  {
    id: "3",
    type: "Revenue",
    amount: 34,
    totalAmount: 1,
    duration: "than yesterday",
  },
  {
    id: "4",
    type: "Followers",
    amount: 91,
    totalAmount: 0,
    duration: "Just updated",
  },
];

export const ANALYTIC_INFO = [
  {
    id: "1",
    photo: "/assets/images/analytics/analytics-info-1.webp",
    name: "Cozy 5 Stars Apartment",
    description: `The place is close to Barceloneta Beach and bus stop just 2 min by walk and near to "Naviglio" where you can enjoy the main night life in Barcelona.`,
    price: 899,
    location: "Barcelona, Spain",
  },
  {
    id: "2",
    photo: "/assets/images/analytics/analytics-info-2.webp",
    name: "Office Studio",
    description: `The place is close to Metro Station and bus stop just 2 min by walk and near to "Naviglio" where you can enjoy the night life in London, UK.`,
    price: 1.119,
    location: "London, UK",
  },
  {
    id: "3",
    photo: "/assets/images/analytics/analytics-info-3.webp",
    name: "Beautiful Castle",
    description: `The place is close to Metro Station and bus stop just 2 min by walk and near to "Naviglio" where you can enjoy the main night life in Milan.`,
    price: 459,
    location: "Milan, Italy",
  },
];

export const WEBSITE_CHART = {
  id: "view-campaign",
  display: "Website Views",
  desc: "Last Campaign Performance",
  modified: "campaign sent 2 days ago",
  data: [
    {
      description: "M",
      sales: 50,
    },
    {
      description: "T",
      sales: 20,
    },
    {
      description: "W",
      sales: 10,
    },
    {
      description: "T",
      sales: 22,
    },
    {
      description: "F",
      sales: 50,
    },
    {
      description: "S",
      sales: 10,
    },
    {
      description: "S",
      sales: 40,
    },
  ],
};
