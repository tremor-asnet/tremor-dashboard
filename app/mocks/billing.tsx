// Constants
import { AGGREGATION_TYPE } from "@/constants";

export const mockSalaryData = [
  {
    type: AGGREGATION_TYPE.PAYPAL,
    value: 45500,
  },
  {
    type: AGGREGATION_TYPE.SALARY,
    value: 2000,
  },
];

export const mockBillingCard = {
  id: 1,
  type: "Mastercard",
  expire: "11/24",
  cardNumber: "4562 1122 4594 7866",
  holderFullName: "Jack Peterson",
  cardLast4Digit: "7866",
};
