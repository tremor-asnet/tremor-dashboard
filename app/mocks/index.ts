export * from "./card";
export * from "./profile";
export * from "./project";
export * from "./charts";
export * from "./analytics";
export * from "./sales";
export * from "./table-list";
export * from "./orders";
export * from "./product";
