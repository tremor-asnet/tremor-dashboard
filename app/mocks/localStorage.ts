// Mock localStorage
export const mockLocalStorage = (() => {
  let store: Record<string, string> = {};

  return {
    getItem: jest.fn((key: string) => store[key] || null),
    setItem: jest.fn((key: string, value: string) => {
      store[key] = value.toString();
    }),
    removeItem: jest.fn((key: string) => {
      delete store[key];
    }),
    clear: jest.fn(() => {
      store = {};
    }),
  };
})();

Object.defineProperty(window, "localStorage", { value: mockLocalStorage });
