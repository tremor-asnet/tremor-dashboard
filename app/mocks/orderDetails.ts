export const mockInvoiceHeader = {
  id: 241342,
  createdAt: "2024-01-11T01:48:52Z",
  orderCode: "KF332",
};

export const mockOderSummary = {
  productPrice: 90,
  delivery: 14,
  taxes: 1.95,
};
