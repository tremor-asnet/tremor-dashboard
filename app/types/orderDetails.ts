export interface InvoiceHeaderProps {
  id: number;
  createdAt: string;
  orderCode: string;
}

export interface OrderSummaryProps {
  productPrice: number;
  delivery: number;
  taxes: number;
}
