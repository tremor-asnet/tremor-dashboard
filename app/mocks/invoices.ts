import { Invoice } from "@/types";

export const MOCK_INVOICES: Invoice[] = [
  {
    date: "March, 21, 2021",
    code: "MS-123456",
    price: 180,
  },
  {
    date: "Apr, 22, 2022",
    code: "MS-123457",
    price: 180,
  },
  {
    date: "May, 23, 2023",
    code: "MS-999999",
    price: 180,
  },
  {
    date: "Jun, 24, 2024",
    code: "MS-888888",
    price: 180,
  },
  {
    date: "Jul, 25, 2025",
    code: "MS-476543",
    price: 180,
  },
];
