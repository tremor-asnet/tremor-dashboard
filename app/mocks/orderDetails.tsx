import { FaBell } from "react-icons/fa";
import { MdInventory } from "react-icons/md";
import { IoCart } from "react-icons/io5";
import { StepOrderProps } from "@/types/orderDetail";

export const mockInvoiceHeader = {
  id: 241342,
  createdAt: "2024-01-11T01:48:52Z",
  orderCode: "KF332",
};

export const mockTrackOrder: StepOrderProps[] = [
  {
    iconInfo: <FaBell />,
    titleInfo: "Order received",
  },
  {
    iconInfo: <MdInventory />,
    titleInfo: "Generate order id",
  },
  {
    iconInfo: <IoCart />,
    titleInfo: "Order transmited to courier",
  },
];
