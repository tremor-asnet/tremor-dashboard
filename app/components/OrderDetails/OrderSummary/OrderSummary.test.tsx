import { render, screen } from "@testing-library/react";
import "@testing-library/jest-dom";

// Components
import { OrderSummary } from ".";

// Mocks
import { mockOderSummary } from "@/mocks/orderDetails";

describe("Test OderSummary component", () => {
  test("Should render correctly", () => {
    const result = render(<OrderSummary {...mockOderSummary} />);
    expect(result).toBeTruthy();
  });

  test("Should match snapshot", () => {
    const result = render(<OrderSummary {...mockOderSummary} />);
    expect(result).toMatchSnapshot();
  });

  test("Should render correct product price", () => {
    render(<OrderSummary {...mockOderSummary} />);
    const result = screen.getByText(
      (_, element) => element?.textContent === "$90",
    );
    expect(result).toBeTruthy();
  });

  test("Should render correct delivery fee", () => {
    render(<OrderSummary {...mockOderSummary} />);
    const result = screen.getByText(
      (_, element) => element?.textContent === "$14",
    );
    expect(result).toBeTruthy();
  });

  test("Should render correct taxes", () => {
    render(<OrderSummary {...mockOderSummary} />);
    const result = screen.getByText(
      (_, element) => element?.textContent === "$1.95",
    );
    expect(result).toBeTruthy();
  });

  test("Should render correct total price", () => {
    render(<OrderSummary {...mockOderSummary} />);
    const result = screen.getByText(
      (_, element) => element?.textContent === "$105.95",
    );
    expect(result).toBeTruthy();
  });
});
