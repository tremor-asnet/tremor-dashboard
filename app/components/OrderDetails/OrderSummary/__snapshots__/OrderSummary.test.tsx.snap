// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test OderSummary component Should match snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <b
        class="text-inherit font-bold dark:text-white"
      >
        Order Summary
      </b>
      <div
        class="tremor-Flex-root flex w-full flex-col justify-between items-center mt-2"
      >
        <div
          class="tremor-Flex-root flex w-full flex-row justify-between items-center"
        >
          <p
            class="text-tremor-default text-tremor-content dark:text-lighter"
          >
            Product Price:
          </p>
          <p
            class="font-bold dark:text-white"
          >
            $
            90
          </p>
        </div>
        <div
          class="tremor-Flex-root flex w-full flex-row justify-between items-center"
        >
          <p
            class="text-tremor-default text-tremor-content dark:text-lighter"
          >
            Delivery:
          </p>
          <p
            class="font-bold dark:text-white"
          >
            $
            14
          </p>
        </div>
        <div
          class="tremor-Flex-root flex w-full flex-row justify-between items-center"
        >
          <p
            class="text-tremor-default text-tremor-content dark:text-lighter"
          >
            Taxes:
          </p>
          <p
            class="font-bold dark:text-white"
          >
            $
            1.95
          </p>
        </div>
        <div
          class="tremor-Flex-root flex w-full flex-row justify-between items-center mt-4"
        >
          <p
            class="text-tremor-content text-xl font-light dark:text-lighter"
          >
            Total:
          </p>
          <p
            class="font-bold dark:text-white"
          >
            $
            105.95
          </p>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <b
      class="text-inherit font-bold dark:text-white"
    >
      Order Summary
    </b>
    <div
      class="tremor-Flex-root flex w-full flex-col justify-between items-center mt-2"
    >
      <div
        class="tremor-Flex-root flex w-full flex-row justify-between items-center"
      >
        <p
          class="text-tremor-default text-tremor-content dark:text-lighter"
        >
          Product Price:
        </p>
        <p
          class="font-bold dark:text-white"
        >
          $
          90
        </p>
      </div>
      <div
        class="tremor-Flex-root flex w-full flex-row justify-between items-center"
      >
        <p
          class="text-tremor-default text-tremor-content dark:text-lighter"
        >
          Delivery:
        </p>
        <p
          class="font-bold dark:text-white"
        >
          $
          14
        </p>
      </div>
      <div
        class="tremor-Flex-root flex w-full flex-row justify-between items-center"
      >
        <p
          class="text-tremor-default text-tremor-content dark:text-lighter"
        >
          Taxes:
        </p>
        <p
          class="font-bold dark:text-white"
        >
          $
          1.95
        </p>
      </div>
      <div
        class="tremor-Flex-root flex w-full flex-row justify-between items-center mt-4"
      >
        <p
          class="text-tremor-content text-xl font-light dark:text-lighter"
        >
          Total:
        </p>
        <p
          class="font-bold dark:text-white"
        >
          $
          105.95
        </p>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
